src/content.entry.ts

```diff
@@ -325,19 +325,53 @@ const resultPromise = (async (): Promise<{
   note: string
   rawLength: number | null
 }> => {
-  const originalPreElement = (() => {
+  if (document.title)
+    return { formatted: false, note: 'document.title is contentful', rawLength: null }
+
+  const originalPreResult:
+    | { kind: 'success', element: HTMLPreElement }
+    | { kind: 'failure', note: string }
+  = (() => {
+    let pre: HTMLPreElement | null = null
     const bodyChildren = document.body.children
     const length = bodyChildren.length
     for (let i = 0; i < length; i++) {
       const child = bodyChildren[i]
-      if (child.tagName === 'PRE') return child as HTMLPreElement
+
+      switch (child.tagName) {
+        case 'PRE': {
+          if (pre != null)
+            return { kind: 'failure', note: 'Multiple body>pre elements' }
+          pre = child as HTMLPreElement
+          break
+        }
+        case 'P':
+        case 'H1':
+        case 'H2':
+        case 'H3':
+        case 'H4':
+        case 'H5':
+        case 'H6': {
+          return { kind: 'failure', note: 'body contains textual elements' }
+        }
+      }
+    }
+
+    if (pre != null) {
+      if (pre.checkVisibility?.() === false)
```

Comment:
`checkVisibility` is baseline 2024, so safer to use optional chaining in case of outdated browser versions